atomic_reactor/plugins/post_generate_maven_metadata.py

```diff
@@ -56,6 +66,101 @@ def get_nvr_components(self, nvr_requests):
 
         return components
 
+    def download_sources(self, url_requests):
+        download_queue = []
+        source_url_to_artifacts = {}
+        remote_source_files = []
+        no_source_artifacts = []
+        session = util.get_retrying_requests_session()
+
+        for url_request in url_requests:
+            artifact_checksums = {algo: url_request[algo] for algo in
+                                  hashlib.algorithms_guaranteed
+                                  if algo in url_request}
+
+            artifact = {
+                'url': url_request['url'],
+                'checksums': artifact_checksums,
+                'filename': os.path.basename(url_request['url'])
+            }
+
+            if 'source-url' not in url_request:
+                no_source_artifacts.append(artifact)
```

Comment:
can you handle this outside download function? Download function should just download instead, no_source_artifacts should be hadnled outside

Reply:
Updated. This part will now be done in the new process_url_requests function.